azure-quantum/azure/quantum/optimization/solvers.py

```diff
@@ -92,66 +93,53 @@ def submit(
             Whether or not to compress the problem when uploading it
             the Blob Storage.
         """
-        # Create a container URL:
-        job_id = Job.create_job_id()
-        logger.info(f"Submitting job with id: {job_id}")
-
-        container_name = f"job-{job_id}"
-
-        if not self.workspace.storage:
-            # No storage account is passed, in this
-            # case, get linked account from the service
-            container_uri = self.workspace._get_linked_storage_sas_uri(
-                container_name
-            )
-            container_client = ContainerClient.from_container_url(
-                container_uri
-            )
-            create_container_using_client(container_client)
-            container_uri = azure.quantum.storage.remove_sas_token(
-                container_uri
-            )
-        else:
-            # Storage account is passed, use it to generate a container_uri
-            container_uri = azure.quantum.storage.get_container_uri(
-                self.workspace.storage, container_name
-            )
-
-        logger.debug(f"Container URI: {container_uri}")
-
-        if isinstance(problem, str):
-            name = "Optimization problem"
-            problem_uri = problem
-        elif isinstance(problem, Problem):
-            name = problem.name
-            problem_uri = problem.upload(
-                self.workspace,
-                compress=compress,
-                container_name=container_name,
+        if isinstance(problem, Problem):
+            # Create job from input data
+            blob = problem.to_blob(compress=compress)
```

Comment:
i'm thinking that the `to_blob` should return a tuple or some other datastructure that indicates not only the payload, but also the `content_type`, `encoding` and maybe even the `input_data_type` as all that information is tightly coupled.
On a similar way, maybe instead of accepting just a storage_uri as parameter, the method accepts the same datastructure?

Reply:
I'm not sure if `to_blob` is the right place for that data. All that method does is convert a payload to a `bytes` object.
I think that the content type, encoding and input data type should all be properties of the `Problem`.